### PR TITLE
v10.0.0 changelog

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,44 @@
...
 
-# 10.0.0-rc.9 September 22, 2021
+# 10.0.0 October 6, 2021
 
-**The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**
+**The Mapbox Maps SDK for Android is now ready for production use.**
+
+## Breaking changes ⚠️
+* Add `@JvmOverloads` where applicable to provide better experience for Java users. ([#656](https://github.com/mapbox/mapbox-maps-android/pull/656))
+* Refactor gestures configuration options to be aligned better across platforms. ([#672](https://github.com/mapbox/mapbox-maps-android/pull/672))
```
Comment (pengdev) [inline CHANGELOG.md:11]: shall we include the list of gestures changes in the change log? or is it better to be in the release notes?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,9 +2,44 @@
...
+* Apply geojson data (using data, url, feature, featureCollection, geometry functions) is fully async now. ([#699](https://github.com/mapbox/mapbox-maps-android/pull/699))
+* Update `getLayerAs` function to return nullable Layer type. ([#673](https://github.com/mapbox/mapbox-maps-android/pull/673))
+* Update map events data models. ([#712](https://github.com/mapbox/mapbox-maps-android/pull/712))
+* Refactor MapEvents listeners, so that each listener will include one event data property. ([#718](https://github.com/mapbox/mapbox-maps-android/pull/718))
```
Comment (pengdev) [inline CHANGELOG.md:15]: same for these two PRs, we have a more detailed list of changes, would be good to include them to at least in release notes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,9 +2,44 @@
...
+* Some abstract classes like `Observer()`, `CustomLayerHost()`, `MapClient()`, `OfflineRegionObserver()` have become interfaces. ([#697](https://github.com/mapbox/mapbox-maps-android/pull/697))
+
+## Features ✨ and improvements 🏁
+* Throw exception when gestures plugin functionality is used but plugin was not created. ([#653](https://github.com/mapbox/mapbox-maps-android/pull/653))
+* Throw exception when camera plugin functionality is used but plugin was not created. ([#668](https://github.com/mapbox/mapbox-maps-android/pull/668))
+* Introduce 3D globe. ([#667](https://github.com/mapbox/mapbox-maps-android/pull/667))
```
Comment (pengdev) [inline CHANGELOG.md:21]: experimental?
```suggestion
* Introduce experimental 3D globe. ([#667](https://github.com/mapbox/mapbox-maps-android/pull/667))
```

Comment (kiryldz) [reply CHANGELOG.md:21]: I'd not state it here, I'll leave it up to decide for @knov

**Review by ank27 — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,9 +2,43 @@
...
```
Comment (knov) [inline CHANGELOG.md:21]:
```suggestion
* Introduce 3D globe (experimental). ([#667](https://github.com/mapbox/mapbox-maps-android/pull/667))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,9 +2,43 @@
...
+* Fix context leak in `LocationProviderImpl`. ([#690](https://github.com/mapbox/mapbox-maps-android/pull/690))
+* Fix native memory leak by explicitly nulling map reference from renderer. ([#687](https://github.com/mapbox/mapbox-maps-android/pull/687))
+* Fix wrong attribute reference in runtime exception text when token is missing. ([#708](https://github.com/mapbox/mapbox-maps-android/pull/708))
+* Fix applying position property to scale bar plugin. ([#677](https://github.com/mapbox/mapbox-maps-android/pull/677))
+* Fix initialisation location puck when no style loaded from code by changing `Plugin#onStart()` call after style loaded started. ([#680](https://github.com/mapbox/mapbox-maps-android/pull/680))
+* Fix attribution/logo jumble when RTL layout is configured. ([#674](https://github.com/mapbox/mapbox-maps-android/pull/674))
+* Fix rendering when using model layer opacity parameter. ([#697](https://github.com/mapbox/mapbox-maps-android/pull/697))
+* Fix not to relayout cached tiles reducing CPU usage when zooming in / out. ([#697](https://github.com/mapbox/mapbox-maps-android/pull/697))
```
Comment (alexshalamov) [inline CHANGELOG.md:35]: > Fix not to relayout cached tiles reducing CPU usage when zooming in / out.

Improve rendering performance by avoiding unnecessary re-layout for cached tiles.

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,9 +2,43 @@
...
+* Fix map render deadlock on Android 8 on power on button. ([#688](https://github.com/mapbox/mapbox-maps-android/pull/688))
+* Fix context leak in `LocationProviderImpl`. ([#690](https://github.com/mapbox/mapbox-maps-android/pull/690))
+* Fix native memory leak by explicitly nulling map reference from renderer. ([#687](https://github.com/mapbox/mapbox-maps-android/pull/687))
+* Fix wrong attribute reference in runtime exception text when token is missing. ([#708](https://github.com/mapbox/mapbox-maps-android/pull/708))
+* Fix applying position property to scale bar plugin. ([#677](https://github.com/mapbox/mapbox-maps-android/pull/677))
+* Fix initialisation location puck when no style loaded from code by changing `Plugin#onStart()` call after style loaded started. ([#680](https://github.com/mapbox/mapbox-maps-android/pull/680))
+* Fix attribution/logo jumble when RTL layout is configured. ([#674](https://github.com/mapbox/mapbox-maps-android/pull/674))
+* Fix rendering when using model layer opacity parameter. ([#697](https://github.com/mapbox/mapbox-maps-android/pull/697))
```
Comment (alexshalamov) [inline CHANGELOG.md:34]: Fix rendering artifacts for a model layer when `model-opacity` property is used.

**Review by knov — APPROVED** [review]

**Review by alexshalamov — APPROVED** [review]

**Pull request merged**